src/providers/cortex/trulens/providers/cortex/endpoint.py

```diff
@@ -17,6 +19,38 @@
 class CortexCallback(EndpointCallback):
     model_config: ClassVar[dict] = dict(arbitrary_types_allowed=True)
 
+    # TODO (Daniel): cost tracking for Cortex finetuned models is not yet implemented.
+
+    def _compute_credits_consumed(
+        self, cortex_model_name: str, n_tokens: int
+    ) -> float:
+        try:
+            model_costs = {}
+            # the credit consumption table needs to be kept up-to-date with
+            # the latest cost information https://www.snowflake.com/legal-files/CreditConsumptionTable.pdf#page=9.
+
+            with open(
+                os.path.join(
+                    os.path.dirname(os.path.realpath(__file__)),
+                    "config/cortex_model_costs.json",
+                ),
+                "r",
```

Comment:
Might be better to just open this file during initialization? And make model_costs a object variable. Or you could also do this lazy by checking if self.model_costs is intiialized.